### ISSUE TITLE
m3u8 playlist - inconsistent re-encoding when fetching metadata for non-ASCII symbols, leading to error

### ISSUE THREAD

**Issue by Domini** [iina/iina#5231]
### IINA Version

1.3.5 Build 141 mpv 0.35.0-419-gf79 FFmpeg 6.0

### Release Channel

Release

### macOS Version

15.0.1

### Steps to Reproduce

Let's assume you have a m3u8 playlist like that
```
#EXTM3U
#EXTINF:,t
http://host/й.mkv

```
(symbol й is https://en.wikipedia.org/wiki/Short_I_(Cyrillic))

If you hexdump the file, й is encoded as two-byte D0 B9 as expected.

When you play the playlist (e.g. double-click in Finder), the entry plays alright. Web-server receives request GET /%D0%B9.mkv as expected.

If you show the playlist as a list (e.g. Playlist - Show Playlist Panel), IINA tries to get metadata for the entry (e.g. to show the length of the media file in the Playlist Panel). Web-server receives request GET /%D0%B8%CC%86.mkv which leads to 404 Not Found and no metadata available.

I think these issues might be related https://github.com/iina/iina/issues/4682, https://github.com/iina/iina/issues/4091.

### Expected Behavior

At the very least, behaviour should be consistent between playing the file and requesting metadata for it. The same (re-)encoding should be used.

And I think the behaviour when playing the file is the right one, as it relays byte to byte what is written in the m3u8 file, as it should be.

### Actual Behavior

IINA re-encodes (normalizes?) UTF8 ASCII>128 symbol that should be two-byte into four-byte (two-byte Cyrillic I + two-byte Unicode accent(?) converting it to Short I).

### How often can you reproduce the issue

always

### Required Actions

- [X] I tried my best to search from the existing issues (including closed issues) and didn't find duplicated issues.

### Optional Actions

- [ ] I tested with the Nightly build and confirmed the issue is not fixed.
- [ ] I tested with the latest version of mpv and confirmed mpv doesn't have this issue.

**Comment by Domini** [comment]
Similar situation with ё (https://en.wikipedia.org/wiki/Yo_(Cyrillic)) - %D1%91 -> %D0%B5%CC%88.

**Comment by svobs** [comment]
You are correct - this looks like #4091. That has been traced to an upstream problem: https://github.com/mpv-player/mpv/issues/10911. As noted in the issue, it has been fixed upstream and is expected to be fixed in the next IINA release. 

I investigated #4682 a while back and it is unrelated. That one was the result of file metadata being saved into the incorrect field.

**Comment by Domini** [comment]
Hey @svobs, thanks for the information. 

I'm not sure, however, I think https://github.com/mpv-player/mpv/issues/10911 addresses a different issue. Here the playlist is already in UTF-8 and for some reason, it's recoded - legal UTF-8 D0 B9 becomes legal UTF-8 D0 B8 CC 86. I want to say "normalized", however, as far as understand, normalization in this case would be the opposite process - converting D0 B8 CC 86 to D0 B9.

Update: ignore the next paragraph. Current IINA running mpv 0.35.0. First mpv with https://github.com/mpv-player/mpv/issues/10911 merged is 0.37.0 (git tag --contains acac614).

Also, upstream https://github.com/mpv-player/mpv/issues/10911 appears to have been merged more than 12 months ago. I'm running the latest available IINA released 4 months ago. Has the changes in the upstream not been propagated to this version yet? Should not the behaviour already be fixed?

**Comment by Domini** [comment]
Hey @svobs,

I hope you are well.

1.4.0 stable is finally out (with MPV 0.38.0 including the aforementioned patch), and unfortunately this issue still persists.

I investigated a little bit further, and it seems that the metadata request from IINA playlist is done within the Lavf library, and not libmpv. User-Agent for playback is "libmpv", but User-Agent for playlist metadata fetching is "Lavf/61.1.100".

Could it be a separate issue?
